libchecktestdata.cc

```diff
@@ -1256,10 +1256,14 @@ void init_checktestdata(std::istream &progstream, int opt_mask)
 	}
 
 	// Initialize random generator
-	struct timespec time;
-	clock_gettime(CLOCK_REALTIME,&time);
-	mpz_class seed = 1000000000 * mpz_class(time.tv_sec) + time.tv_nsec;
-	gmp_rnd.seed(seed);
+	if ( seed == static_cast<unsigned long>(-1)) {
```

Comment:
What about making `seed` a (signed) long? Then this static cast isn't necessary, and also the initial assignment on line 96 is strictly speaking correct, and an uninitialized seed can be distinguished from one specified as `ULONG_MAX`.

Reply:
Yes, completely agree.